crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -155,6 +161,8 @@ pub struct DiskFileSystem {
     #[turbo_tasks(debug_ignore, trace_ignore)]
     dir_invalidator_map: Arc<InvalidatorMap>,
     #[turbo_tasks(debug_ignore, trace_ignore)]
+    authorative_write_map: Arc<Mutex<HashMap<PathBuf, AutoMap<Invalidator, WriterState>>>>,
```

Comment:
Why does this need an inner mapping? Shouldn't this be achievable with just a `Invalidator`? Either you're the first writer, you're the first writer after the previous writer became inactive, or you're in conflict and we can early exit.

Reply:
We want all write operations to report the conflict, including the first one. So in case of a conflict we need to recompute all write operations to report the conflict. But we only want to invalidate write operations that have not reported a conflict yet, since we would end up in a loop otherwise. So that's why we store the state for each operation and only invalidate the authoritive ones.